src/Workspaces/Remote/Core/TestUtils.cs

```diff
@@ -9,23 +16,117 @@ internal static class TestUtils
     {
         public static void RemoveChecksums(this Dictionary<Checksum, object> map, ChecksumWithChildren checksums)
         {
-            map.Remove(checksums.Checksum);
+            var set = new HashSet<Checksum>();
+            set.AppendChecksums(checksums);
 
-            foreach (var child in checksums.Children)
+            RemoveChecksums(map, set);
+        }
+
+        public static void RemoveChecksums(this Dictionary<Checksum, object> map, IEnumerable<Checksum> checksums)
+        {
+            foreach (var checksum in checksums)
+            {
+                map.Remove(checksum);
+            }
+        }
+
+        internal static async Task AssertChecksumsAsync(
+            AssetService assetService,
+            Checksum checksumFromRequest,
+            Solution solutionFromScratch,
+            Solution incrementalSolutionBuilt)
+        {
+#if DEBUG
```

Comment:
📝 It looks like this is unnecessary. The code is only conditionally called.

Reply:
I left it since no need to compile in release but ya, probably don't need i preprocessor. but. don't want to re-start whole test running. since it won't hurt I will leave it as it is. but remove when I have chance to touch this code again next time.